controllers/admin/AdminImportController.php

```diff
@@ -1756,39 +2362,54 @@ protected function productImportOne($info, $default_language_id, $id_lang, $forc
-                        $category_to_create->link_rewrite = AdminImportController::createMultiLangField($category_link_rewrite);
-                        if (($field_error = $category_to_create->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-                            ($lang_field_error = $category_to_create->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true &&
-                            !$validateOnly &&  // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
-                            $category_to_create->add()) {
-                            $product->id_category[] = (int)$category_to_create->id;
+                        $categoryToCreate = new Category();
+                        $categoryToCreate->id = (int)$value;
+                        $categoryToCreate->name = AdminImportController::createMultiLangField($value);
+                        $categoryToCreate->active = 1;
+                        // Default parent is home for unknown category to create
+                        $categoryToCreate->id_parent = Configuration::get('PS_HOME_CATEGORY');
+                        $categoryLinkRewrite = Tools::link_rewrite($categoryToCreate->name[$defaultLanguageId]);
+
+
+                        // FIXME needing such a comment about "&& !$validateOnly" position is a code smell.
+                        // Refacto needed.
+                        if (($fieldError = $categoryToCreate->validateFields(UNFRIENDLY_ERROR, true)) === true
+                            && ($langFieldError = $categoryToCreate->validateFieldsLang(
+                                UNFRIENDLY_ERROR,
+                                true
+                            )) === true
+                            && !$validateOnly // Do not move this condition: previous tests should be played always, but next ->add() test should not be played in validateOnly mode
+                            && $categoryToCreate->add()
+                        ) {
+                            $categoryToCreate->link_rewrite = AdminImportController::createMultiLangField(
+                                $categoryLinkRewrite
+                            );
+                            $product->id_category[] = (int)$categoryToCreate->id;
                         } else {
                             if (!$validateOnly) {
-                                $this->errors[] = sprintf(
-                                    $this->trans('%1$s (ID: %2$s) cannot be saved', array(), 'Admin.Advparameters.Notification'),
-                                    $category_to_create->name[$default_language_id],
-                                    (isset($category_to_create->id) && !empty($category_to_create->id))? $category_to_create->id : 'null'
-                                );
+                                $this->errors[] = $this->trans('%1$s (ID: %2$s) cannot be saved', array(
```

Comment:
Same comment about translation placeholder. Please use a variable name surrounded by `%%` an add it in the second parameter.